_published/0012-authentication.md

```diff
@@ -0,0 +1,167 @@
+interchangeable. If it is used by the public they can set up their own IDP
+and connect to that.
+
+### Authorisation server
+
+There are many [alternatives](https://openid.net/developers/certified/) on
+alternative authorisation servers.
+
+We had a quick info share from spark the spark project. They are using open
+identity server 3 or 4. This however appears to be relatively tightly coupled
+with the .NET framework.
+
+One that seems to align well with this
+project is `Ory - Hydra` [github](https://github.com/ory/hydra)
+and [docs](https://www.ory.sh/hydra/docs/). The descriptions of them from their
+github at the present time is:
+
+```
+ORY Hydra is a hardened, OpenID Certified OAuth 2.0 Server and OpenID Connect
+Provider optimized for low-latency, high throughput, and low resource
+consumption. ORY Hydra is not an identity provider (user sign up, user
+login, password reset flow), but connects to your existing identity provider
+through a login and consent app. Implementing the login and consent app in a
+different language is easy, and exemplary consent apps (Node) and SDKs for all
+common languages are provided.
+```
+They appear to be funded through opencollective and patreon.
+
+Another alternative to consider could be [Keycloak](https://www.keycloak.org/).
+This appears to contain
+much more finished work such as a user admin panel for
+setting user and group rights. It is styled using themes. Con: Its a redhat
+project which means association with IBM which has had a certain downside in
+relation to opensource projects in the past see
+
+## Compatibility
+
+Expect no compatability between versions of wharf without authentication to
+versions of wharf with authentication. There is a potential threat in leaving
+the system backwards compatible in that a non-secured version of the api could
+run together in an otherwise secure cluster.
+
+We should aim to pick a solution where any identity provider (IDP) can be used.
+Locking ourselves to a single IDP when a standard such as OICD exists seems
+unnecessary. Keeping the choice of IDP open will leave more room for
+choice in the future.
+
+## Alternative solutions
+
+The starting assumption for this RFC is usage of the [OpenID Connect spec](https://openid.net/specs/openid-connect-core-1_0.html).
+There is a [list of certified providers](https://openid.net/developers/certified/).
+
+## Future possibilities
+
+After this we should consider
+- User management interfaces
+- User group management interfaces
+- Authorization
+- Permissions
+
+## Unresolved questions
+
+1. Swagger - how will this integrate with OpenID Connect auth?
+   some info is available at [swagger docs](https://swagger.io/docs/specification/authentication/openid-connect-discovery/)
+2. Production like deployments - We need to setup helm charts?, IDP
+   configuration for hydra?, and maybe more. [Some info available in docs](https://www.ory.sh/hydra/docs/production)
```

Comment:
For Postgres we've decided that users deploy this themselves. Maybe we could do the same with hydra, depending on how many ways there are of hosting it. If it's "just" a basic pod, no PVC and no special requirements, then we could probably just embed it into our wharf-helm Helm chart